cdc/owner/ddl_manager.go

```diff
@@ -0,0 +1,480 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package owner
+
+import (
+	"context"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	timodel "github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/puller"
+	"github.com/pingcap/tiflow/cdc/redo"
+	"go.uber.org/zap"
+)
+
+// globalDDLs is the DDLs that affect all tables in the changefeed.
+// we need to wait all tables checkpointTs reach the DDL commitTs
+// before we can execute the DDL.
+var globalDDLs = []timodel.ActionType{
```

Comment:
Should `truncate table` be in `globalDDLs`?

Reply:
I think that when handling a `truncate table` ddl, you only need to stop its `preTable` progress at ddl commiTs before executing it. So there is no need to put it in `globalDDLs`. 
I will do more testing with this ddl.

Reply:
Based on my understanding, the `truncate table` DDL has two table information sets: `preTableInfo` and `tableInfo`. The `ddl_manager` uses the `preTableInfo` to prevent the checkpointTs from progressing, ensuring that the DDL is executed without any disruption. After the DDL is executed, the scheduler will use the `tableID` contained in the `tableInfo`, along with the `commiTs` as `startTs`, to create a new table task. 

So I think that the `truncate table` DDL does not need to be included in `globalDDLs`. However, I will conduct more tests on this DDL to ensure that our understanding is correct. Thank you for reminding me of this important consideration.